docs/src/snap/reference/bootstrap-config-reference.md

```diff
@@ -0,0 +1,422 @@
+# Bootstrap configuration file reference
+
+A YAML file can be supplied to the `k8s bootstrap` command to configure and customize the cluster.
+This Reference section provides the format of this file by listing all available options and their details.
+
+## Format Specification
+
+### cluster-config.network
+**Type:** `object` <br>
+**Required:** `No`
```

Comment:
I think this is a bit unnecessary. 
Just add a remark "required if bgp_enabled" in the description of the few options where this is required.

Reply:
I tried to match with docs from other teams, it's true that we mostly don't have anything required but I also feel like explicitly stating these is worth in a Reference page. I'll leave it up to @evilnick if we should keep these

Reply:
I think there are a number of things we could probably do to improve this. i will make a card to re-evaluate it but for now I think it just needs to exist